pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -115,10 +115,14 @@ pub mod pallet {
 
 		/// The asset id of the collateral currency.
 		type CollateralAssetId: Get<CollateralAssetIdOf<Self>>;
-		/// Who can create new bonded currency pools.
+		/// The origin for most permissionless and priviledged operations.
+		type DefaultOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
+		/// The dedicated origin for creating new bonded currency pools (typically permissionless).
 		type PoolCreateOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
+		/// The origin for permissioned operations (force_* transactions).
+		type ForceOrigin: EnsureOrigin<Self::RuntimeOrigin>;
 		/// The type used for pool ids
-		type PoolId: Parameter + MaxEncodedLen + From<[u8; 32]> + Into<Self::AccountId>;
+		type PoolId: Parameter + MaxEncodedLen + From<[u8; 32]> + Into<[u8; 32]>;
 
```

Comment:
`From<[u8; 32]>` is then not necessary. You can create a `[u8; 32]` with this trait and then call `AccountId::from(<[u8; 32]>)`. `PoolId` is not required to deal with `From<[u8; 32]>`, unless I am missing something.

Reply:
I also can't tell you what that was for, but I assumed it had a purpose. @Ad96el why do we need to be able to create a pool id from a 32 byte string?

Reply:
I am also not sure anymore. We can get rid of the trait bound, I guess.

Reply:
c0722ea8bd2fdc02a1da7ba98e0a852d3069e743